templates/deployment.yml

```diff
@@ -29,11 +29,21 @@ spec:
       - image: armory/agent-k8s:{{ .Chart.AppVersion }}
         imagePullPolicy: IfNotPresent
         name: agent
-        {{- if .Values.cloudEnabled }}
         env:
+        {{- if .Values.cloudEnabled }}
         - name: ARMORY_HUB
           value: "true"
         {{- end }}
+        {{- with .Values.env }}
+        {{- /* Unfortunately this loop can't validate that the map has a name
+               and value structure. If name or value is mispelled, then it will
+               fail silently and create an invalid env structure.
+           */}}
+        {{- range . }}
```

Comment:
I'm unfamiliar with this notation, does it work with latest helm?
How about
```
        {{- range $key, $var := .Values.env }}
              - name: {{ $var.name }}
                 value: {{ $var.value }}
        {{- end }}
```

Reply:
I guess it works because it's inside `{{- with .Values.env }}`:
```
{{- with .Values.env }}
    {{- range . }}
    {{- end }}
{{- end }}
```
Anyway yours looks a bit cleaner, we can use it on the next change to the charts, it just would be `{{- range $index, $var := .Values.env }}` instead of `{{- range $key, $var := .Values.env }}` to avoid confusion using the `key` word.